### ISSUE TITLE
php processes spammed when linting *.blade.php files

### ISSUE THREAD

**Issue by piperone** [SublimeLinter/SublimeLinter-phpmd#15]
Unsure wether this belongs in the SublimeLinter-repo, so apologies in advance if this issue is misplaced.

When linting files from Laravel's template language blade, around 60 php (!) processes are spawned by SublimeLinter. The console spits out the following:

```
SublimeLinter: #1 SublimeLinter.sublime-package:1066: phpmd: myfile.blade.php ['/path/to/my/bin/phpmd '${file}', 'text', 'cleancode,codesize,controversial,design,naming,unusedcode']
SublimeLinter: #1 SublimeLinter.sublime-package:1069: phpmd: cwd: /path/to/my/project
SublimeLinter: #2 SublimeLinter.sublime-package:1066: phpmd: myfile.blade.php ['/path/to/my/bin/phpmd '${file}', 'text', 'cleancode,codesize,controversial,design,naming,unusedcode']
SublimeLinter: #2 SublimeLinter.sublime-package:1069: phpmd: cwd: /path/to/my/project
(...)
SublimeLinter: #63 SublimeLinter.sublime-package:1066: phpmd: myfile.blade.php ['/path/to/my/bin/phpmd '${file}', 'text', 'cleancode,codesize,controversial,design,naming,unusedcode']
SublimeLinter: #63 SublimeLinter.sublime-package:1069: phpmd: cwd: /path/to/my/project
```

This completely bogs down the CPU on my machine, ST becomes unresponsive and I have to shut down ST and let the processes die out. I changed linter-mode from 'background' to 'save' in order to try and nail the problem. Sure enough, when I save a blade-file, this behaviour is displayed.

**Comment by piperone** [comment]
In case it's relevant, here's some version info:

PHPmd: 2.6.0
SublimeLinter: 4.2.1
SublimeLinter-phpmd: 1.1.3
Sublime Text: 3 Build 3143

Running macOS High Sierra 10.13.4

**Comment by kaste** [comment]
You should look at this linter a selector. I don't know blade and which parts of the template you actually can lint with phone, but in case phpmd can lint whole blade files you should probably add 'text.blade' to the selector in your linter settings.

**Comment by braver** [comment]
You can set `text.blade` in the selector setting yourself. If this works for you please report back and perhaps it's a good idea to add it to the default. SublimeLinter 4.3 adds some more features to prevent spamming processes, play around with those too. 
If none of that works, we don't have a lot of blade developers working on SublimeLinter so we need more info to debug. I kinda think one of the options available should work though.

**Comment by piperone** [comment]
So I added a selector to my config like so:
```
"phpmd": {
    "@disable": false,
    "args": ["codesize,controversial,design,naming,unusedcode"],
    "excludes": [],
    "selector": "text.blade"
}
```
Now, a single blade.php-file is linted only once. However, it doesn't really make sense for entire blade-views to be piped to phpmd. So, the problem was that for **every** php-scope detected inside a blade file, phpmd was run once. Large blade-files may contain many, many separate php-scopes and that seems to explain why the php-process was spammed as I described earlier.

Here's a simple example of a blade-file:

```blade
@php
    $foo = 'apple';
    $bar = 'orange';
@endphp

<ul>
    <li>{{ $foo }}</li>
    <li>{{ $bar }}</li>
</ul>

@php
    $fruit = "$foo, $bar";
@endphp
```

The above contains 4 distinct php-scopes and each are run through phpmd. As I said, adding 'text.blade' to selector, solves this because the entire thing is linted as one.

Now, I have a problem, however, because I also want my .php-files to be run through phpmd. So, I must add 'source.php' and/or 'embed.php' to the "selector" setting. The problem then re-occurs because each php-scope is, again, run separately through phpmd as before. I really can't think of a good solution to this, as I have zero experience working with the SublimeLinter framework.

By the way, I'm seeing the [exact same behavior](https://github.com/SublimeLinter/SublimeLinter-phpcs/issues/35) with SublimeLinter-phpcs.
